src/index.js

```diff
@@ -189,46 +189,52 @@ class AudioEngine {
      * Store a reference to it the sound in the audioBuffers dictionary, indexed by soundId
      * @param  {object} sound - an object containing audio data and metadata for a sound
      * @property {Buffer} data - sound data loaded from scratch-storage.
-     * @property {string} format - format type, either empty or adpcm.
      * @returns {?Promise} - a promise which will resolve to the soundId if decoded and stored.
      */
     decodeSound (sound) {
         const soundId = uid();
         let loaderPromise = null;
 
         // Make a copy of the buffer because decoding detaches the original buffer
-        const bufferCopy = sound.data.buffer.slice(0);
-
-        switch (sound.format) {
-        case '':
-            // Check for newer promise-based API
-            if (this.audioContext.decodeAudioData.length === 1) {
-                loaderPromise = this.audioContext.decodeAudioData(bufferCopy);
-            } else {
-                // Fall back to callback API
-                loaderPromise = new Promise((resolve, reject) => {
-                    this.audioContext.decodeAudioData(bufferCopy,
-                        decodedAudio => resolve(decodedAudio),
-                        error => reject(error)
-                    );
-                });
-            }
-            break;
-        case 'adpcm':
-            loaderPromise = (new ADPCMSoundDecoder(this.audioContext)).decode(bufferCopy);
-            break;
-        default:
-            return log.warn('unknown sound format', sound.format);
+        const bufferCopy1 = sound.data.buffer.slice(0);
+
+        // Attempt to decode the sound using the browser's native audio data decoder
+        // Check for newer promise-based API
```

Comment:
I feel like this would make more sense as a helper method out in the module namespace:

```js
// Wrapper to ensure audioContext.decodeAudioData is "promisified"
const decodeAudioData = (context, buffer) => {
  // Detect the Promise based API - it only has one argument to the function
  if (context.decodeAudioData.length === 1) {
    return context.decodeAudioData(buffer);
  }
  return new Promise((resolve, reject) => context.decodeAudioData(buffer, resolve, reject));
};
```

lets you just do `return decodeAudioData(this.audioContext, bufferCopy1).then(...)` in here, a lot cleaner to read.